File: GFG/Tree/childsum.cpp
#include<iostream>
#include"../ds.h"
using namespace std;

bool csum(TNode<int>* root){
    if(root->left==NULL && root->right==NULL){
        return true;
    }
    int sum=0;
    if(root->left!=NULL){sum+=root->left->key;}
    if(root->right!=NULL){sum+=root->right->key;}
    
    return(root->key==sum && csum(root->left) && csum(root->right));
}

int main(){
    TNode<int>* root=new TNode(20);
    root->left=new TNode(8);
    root->left->left=new TNode(3);
    // root->left->left->left=new TNode(15);
    root->left->right=new TNode(5);
    // root->left->left->right=new TNode(28);
    root->right=new TNode(12);
    // root->right->left=new TNode(45);
    // root->right->right=new TNode(60);
    // root->right->right->right=new TNode(70);
    // root->right->right->left=new TNode(55);
    cout<<csum(root);
    return 0;
}

File: GFG/Tree/childsum.py
import ds

def csum(root):
    if(root.left is None and root.right is None):
        return True
    suma=0
    if(root.left!=None):
        suma+=root.left.key
    if(root.right!=None):
        suma+=root.right.key
    return(root.key==suma and csum(root.left) and csum(root.right))
    

root=ds.TNode(40)
root.left=ds.TNode(30)
root.left.left=ds.TNode(25)
root.left.left.left=ds.TNode(15)
root.left.right=ds.TNode(35)
root.left.left.right=ds.TNode(28)
root.right=ds.TNode(50)
root.right.left=ds.TNode(45)
root.right.right=ds.TNode(60)
root.right.right.right=ds.TNode(70)
root.right.right.left=ds.TNode(55)
print(csum(root))